### DISCUSSION TITLE
How to install Clickhouse DB separately

### DISCUSSION THREAD

**Discussion by linh060** [SigNoz/signoz#737] [General]
Hi All,
Can we install Clickhouse DB separately? We really want to install Clickhouse DB on Linux server (not using docker) and configure Signoz software to connect to it. Please consider my suggestion and provide this option in the installation method.
Thanks so much!

**Comment by pranay01** [comment]
@linh060 Trying to understand the use case better. Is this primarily because you already run an instance of Clickhouse which you want to send to connect with SIgNoz?

**Reply by linh060** [reply]
@pranay01 , Yes, exactly:
As You think, We had a Clickhouse DB before, and We want to using it for Signoz.
Additionally, We want fully manage Clickhouse DB and can intergrate data with some another solutions (as Bigdata and Data analytic, ...).

**Reply by pranay01** [reply]
got it

**Reply by pranay01** [reply]
@linh060 So we have not tested this, but this should be doable. You have to change your clickhouse destination in 5 places, and remove clickhouse from docker-compose. 

Step wise:

1. Change clickhouse destination in following files, and point to your installation
https://github.com/SigNoz/signoz/blob/develop/deploy/docker/clickhouse-setup/prometheus.yml#L26
https://github.com/SigNoz/signoz/blob/develop/deploy/docker/clickhouse-setup/docker-compose.yaml#L37
https://github.com/SigNoz/signoz/blob/develop/deploy/docker/clickhouse-setup/otel-collector-config.yaml#L47
https://github.com/SigNoz/signoz/blob/develop/deploy/docker/clickhouse-setup/otel-collector-config.yaml#L49
https://github.com/SigNoz/signoz/blob/develop/deploy/docker/clickhouse-setup/otel-collector-metrics-config.yaml#L36

2. Remove clickhouse service section from docker-compose
https://github.com/SigNoz/signoz/blob/develop/deploy/docker/clickhouse-setup/docker-compose.yaml#L4-L14

3. And then docker compose restart
https://signoz.io/docs/operate/docker-standalone/#stopstart

Would be curious to know if the above works or not, and then we can add in our docs also :) if some other users want something similar

**Reply by wang-qijia** [reply]
The above modification scheme is feasible. I would like to ask here does ClickHouse support distributed clusters? How to support it? thanks

**Reply by ankitnayan** [reply]
@rrsunhome yes it does and we have started to work on that. But for time being https://medium.com/@merticariug/distributed-clickhouse-configuration-d412c211687c

**Comment by zriyansh** [comment]
@linh060 any update on this? Did it work?

**Comment by TheShubhendra** [comment]
Hi @pranay01 and @ankitnayan I have some similar requirement
I installed a single-node ClickHouse server and encountered issues migrating existing logs and traces. Initially, I got the error: No macro 'shard' in config while processing substitutions in '/clickhouse/tables/{uuid}/{shard}' while creating signoz_logs.logs_v2_resource. To resolve this, I set the macros shard and replica to 1, but then I encountered another error: Can't create replicated table without ZooKeeper (Code: 225). Do you have any suggestions on migrating those distributed tables to single node deployment? We don't have much load as of now so to reduce complexity and overhead don't want to go with cluster deployment.
Also signoz will run with single node clickhouse or not?
